test/main.c

```diff
@@ -530,6 +528,96 @@ ConvertToRgba32(SDL_Surface **surface_p)
     return SDL_TRUE;
 }
 
+static void
+DumpPixels(const char *filename, SDL_Surface *surface)
+{
+    const unsigned char *pixels = surface->pixels;
+    const unsigned char *p;
+    size_t w, h, pitch;
+    size_t i, j;
+
+    fprintf(stderr, "%s:\n", filename);
+
+    if (surface->format->palette) {
+        size_t n = surface->format->palette->ncolors;
+
+        if (n < 0)
```

Comment:
this check can be removed here - `n` can't be negative.

Reply:
It should be a check on `surface->format->palette->ncolors` before we assign it to `n`, ideally.

Reply:
Let's fix it that way, then